src/main/java/org/spongepowered/mod/mixin/core/server/MixinNetHandlerPlayServer.java

```diff
@@ -82,4 +93,14 @@ public void sendCustomPayload(Object plugin, String channel, byte[] data) {
         sendPacket(new S3FPacketCustomPayload(channel, new PacketBuffer(Unpooled.wrappedBuffer(data))));
     }
 
+    @Inject(method = "processVanilla250Packet", at = @At(value = "INVOKE", target = "checkThreadAndEnqueue", shift = At.Shift.AFTER),
+            cancellable = true)
+    public void onProcessCustomPacket(C17PacketCustomPayload packetIn, CallbackInfo ci) {
+        if (IMixinServer.vanillaChannels.contains(packetIn.getChannelName())) {
+            return;
+        }
+        ci.cancel();
```

Comment:
Whoa there, so what you're saying here is that if a plugin registers an channel then **no other consumers** get to handle it? There is **no way** this is an acceptable behaviour.